packages/swarm-components/src/Dropdown.jsx

```diff
@@ -0,0 +1,462 @@
+			component: Comp,
+			as: AsComp = 'a',
+			style,
+			setState,
+			state,
+			index,
+			_ref,
+			...props
+		},
+		ref
+	) => {
+		const Link = Comp || AsComp;
+		if (Comp) {
+			console.warn(
+				'[@reach/menu-button]: Please use the `as` prop instead of `component`.'
+			);
+		}
+		return (
+			<MenuItem role="none" index={index} onSelect={k} _ref={k}>
+				<Link
+					role="menuitem"
+					data-reach-menu-item
+					tabIndex="-1"
+					data-selected={index === state.selectionIndex ? true : undefined}
+					onClick={wrapEvent(onClick, event => {
+						setState({ ...state, ...close() });
+					})}
+					onKeyDown={wrapEvent(onKeyDown, event => {
+						if (event.key === 'Enter') {
+							// prevent MenuItem's preventDefault from firing,
+							// allowing this link to work w/ the keyboard
+							event.stopPropagation();
+						}
+					})}
+					ref={node => {
+						assignRef(_ref, node);
+						assignRef(ref, node);
+					}}
+					style={{ ...style }}
+					{...props}
+				/>
+			</MenuItem>
+		);
+	}
+);
+
+MenuLink.propTypes = {
+	onKeyDown: func,
+	onClick: func,
+	component: any,
+	as: any,
+	style: object,
+	setState: func,
+	state: object,
+	index: number,
+	_ref: func,
+};
+///////////////////////////////////////////////////////////////////
+
+let MenuList = React.forwardRef((props, ref) => {
+	const { state, setState, refs } = useContext(MenuContext);
+	const menuRef = useRef(null);
+
+	return (
+		state.isOpen && (
+			<Portal>
+				<WindowSize>
+					{() => (
+						<Rect>
+							{({ rect: menuRect, ref: menuRef }) => (
+								<div
+									data-reach-menu
+									ref={menuRef}
+									style={getStyles(state.buttonRect, menuRect)}
+								>
+									<MenuListImpl {...props} ref={ref} />
+								</div>
+							)}
+						</Rect>
+					)}
+				</WindowSize>
+			</Portal>
+		)
+	);
+});
+
+MenuList.propTypes = {
+	children: node,
+};
+
+let isFocusableChildType = (child, types = []) => types.includes(child.type);
+let getFocusableMenuChildren = (children, types) => {
+	let focusable = [];
+	Children.forEach(children, child => {
+		if (isFocusableChildType(child, types)) focusable.push(child);
+	});
+	return focusable;
+};
+
+let MenuListImpl = React.forwardRef(
+	(
+		{
+			children,
+			onKeyDown,
+			onBlur,
+			focusableChildrenTypes = [MenuItem, MenuLink],
+			...rest
+		},
+		ref
+	) => {
+		const { state, setState, refs } = useContext(MenuContext);
+		let focusableChildren = getFocusableMenuChildren(
+			children,
+			focusableChildrenTypes
+		);
+		return (
+			<div
+				data-reach-menu-list
+				{...rest}
+				role="menu"
+				aria-labelledby={state.buttonId}
+				tabIndex="-1"
+				ref={node => {
+					refs.menu = node;
+					assignRef(ref, node);
+				}}
+				onBlur={event => {
+					if (!state.closingWithClick && !refs.menu.contains(event.relatedTarget)) {
+						setState({ ...state, ...close() });
+					}
+				}}
+				onKeyDown={wrapEvent(onKeyDown, event => {
+					if (event.key === 'Escape') {
```

Comment:
no biggie, `switch/case` would work pretty well here instead of sequence of `if/else`